boringtun/src/device/mod.rs

```diff
@@ -830,18 +829,12 @@ impl Device {
                         Some(peer) => peer,
                     };
 
-                    let mut p = peer.lock();
-
-                    let mut public_key = String::with_capacity(64);
-                    for byte in p.tunnel.peer_static_public().as_bytes() {
-                        public_key.push_str(&format!("{:02X}", byte));
-                    }
-                    // We found a peer, use it to decapsulate the message+
                     let mut flush = false; // Are there packets to send from the queue?
-                    match p
-                        .tunnel
-                        .handle_verified_packet(parsed_packet, &mut t.dst_buf[..])
-                    {
+                    let res = {
+                        let mut tun = peer.tunnel.lock();
+                        tun.handle_verified_packet(parsed_packet, &mut t.dst_buf[..])
+                    };
```

Comment:
nit: it should be equally correct yet more readable to write:
```rust
let res = peer.tunnel.lock().handle_verified_packet(parsed_packet, &mutt.dst_buf[..]);
```
this applies to some other places also.